lib/cf_deployer/driver/cloud_formation_driver.rb

```diff
@@ -21,13 +21,18 @@ def update_stack template, opts
           CfDeployer::Driver::DryRun.guard "Skipping update_stack" do
             aws_stack.update opts.merge(:template => template)
           end
+
+          return !CfDeployer::Driver::DryRun.enabled?
         rescue AWS::CloudFormation::Errors::ValidationError => e
           if e.message =~ /No updates are to be performed/
             Log.info e.message
+            return false
           else
             raise
           end
         end
+
+        true
```

Comment:
Good catch.  I think I initially had it default to true and did *not* have the dry-run guard written with a return.  My original intent was, by default, it should be assumed to have succeeded unless proven otherwise.  

One such case to get to that line of code would be to have an exception that is not a "No updates are to be performed".  I don't know if that would or would not have performed an update, but I'm not sure how to distinguish.  In that case, I'd rather it behave as it currently does (which is to check the status of the stack).